File: Data Structures & Algorithms/permutations/submission-0.py
class Solution:
    def permute(self, nums: List[int]) -> List[List[int]]:
        res = []
        n = len(nums)

        def f(idx):
            if idx == n:
                res.append(nums[:])
                return 

            for i in range(idx, n):
                nums[idx], nums[i] = nums[i], nums[idx]
                f(idx + 1)
                nums[idx], nums[i] = nums[i], nums[idx]

        f(0)
        return res

File: Data Structures & Algorithms/move-zeroes/submission-0.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int po = 0;

        for(int i = 0; i < nums.size(); i++){
            if(nums[i] != 0){
                nums[po] = nums[i];
                po++;
            }
        }
        while( po < nums.size()){
            nums[po] = 0;
            po++;
        }        
        
    }
};

/*
        nums = [0,0,1,2,0,5]
                    po
                           i
        nums[pos] = nums[i] = nums= [1,0,1,2,0,5]
                              nums = [1,2,1,2,0,5]
                              nums = [1,2,5,2,0,5]
                              nums = [1,2,5,0,0,0]

        */